aten/src/ATen/WrapDimUtils.h

```diff
@@ -30,14 +30,15 @@ static inline int64_t maybe_wrap_dim(int64_t dim, const std::vector<std::vector<
   return maybe_wrap_dim(dim, tensor_sizes[0].size());
 }
 
-// wrap each of dims basing on dim_post_expr
-static inline void maybe_wrap_dims(std::vector<int64_t>& dims, int64_t dim_post_expr) {
+static inline void maybe_wrap_dims_n(int64_t * dims, int64_t ndims, int64_t dim_post_expr) {
+  // wrap each of dims basing on dim_post_expr
   if (dim_post_expr <= 0) {
     dim_post_expr = 1; // this will make range [-1, 0]
   }
   int64_t min = -dim_post_expr;
   int64_t max = dim_post_expr - 1;
-  for (auto& dim : dims) {
+  for (int64_t i = 0; i < ndims; ++i) {
```

Comment:
Why make this change?

Reply:
To support passing `DimVector` in place of `std::vector`.